Доработка апи (#1)

Arturchel174 opened the pull request:
* Update misc.go

CallMethod - универсальный метод для вызова API Bitrix24

Если используется вебхук (auth начинается с http)

* Update b24.go

Для вебхуков проверяем формат

Arturchel174 committed:
Доработка апи (#1)

* Update misc.go

CallMethod - универсальный метод для вызова API Bitrix24

Если используется вебхук (auth начинается с http)

* Update b24.go

Для вебхуков проверяем формат

lanyere commented:
Добрый день!
Спасибо за PR и вклад в библиотеку, мы это очень ценим! ^_^

Идея добавить входящие вебхуки хорошая :)

Рекомендую внести правки следующим образом для целостности и единообразия кода:

1. Добавить другой метод в файле **b24.go** - NewWebhook(url string) *API , который проверит валидность ссылки
_(для предотвращения использования NewAPI, которое требует clientID и clientSecret, также это сделает код чистым)_
2. Добавить структуру **API** параметр WebhookURL string
_(чтобы не задевать Auth и Domain, да и будет чётко обозначено и облегчит поиск для пользователей библиотеки)_
3. Изменить CallMethod на **CustomMethod**
_(потому, что есть callMethod и схожее название может вызвать путаницу, custom обозначит, что это для всех запросов, которые ещё не предусмотрены)_

lanyere commented:
Ещё, в buildURL можно будет просто добавить if (api.WebhookURL != "" { return api.WebhookURL } , потому, что ссылка уже будет проверена на валидность при NewWebhook, там объединять ничего не нужно будет, полагаю

Arturchel174 committed:
правки для целостности и единообразия кода

1) Добавить другой метод в файле b24.go - NewWebhook(url string) *API , который проверит валидность ссылки
(для предотвращения использования NewAPI, которое требует clientID и clientSecret, также это сделает код чистым)
2) Добавить структуру API параметр WebhookURL string
(чтобы не задевать Auth и Domain, да и будет чётко обозначено и облегчит поиск для пользователей библиотеки)
3) Изменить CallMethod на CustomMethod
(потому, что есть callMethod и схожее название может вызвать путаницу, custom обозначит, что это для всех запросов, которые ещё не предусмотрены)
4) buildURL можно будет просто добавить if (api.WebhookURL != "" { return api.WebhookURL }

Arturchel174 commented:
Здравствуйте, внес правки!

lanyere committed:
Merge branch 'main' into main
